instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractor.java

```diff
@@ -54,7 +54,46 @@ public static <REQUEST, RESPONSE> HttpClientAttributesExtractorBuilder<REQUEST,
   @Override
   public void onStart(AttributesBuilder attributes, Context parentContext, REQUEST request) {
     super.onStart(attributes, parentContext, request);
-    internalSet(attributes, SemanticAttributes.HTTP_URL, getter.url(request));
+    internalSet(attributes, SemanticAttributes.HTTP_URL, stripSensitiveData(getter.url(request)));
+  }
+
+  @Nullable
+  private static String stripSensitiveData(@Nullable String url) {
+    if (url == null || url.isEmpty()) {
+      return url;
+    }
+
+    int schemeEndIndex = url.indexOf(':');
+
+    if (schemeEndIndex == -1) {
+      // not a valid url
+      return url;
+    }
+
+    int len = url.length();
+    if (len <= schemeEndIndex + 2
+        || url.charAt(schemeEndIndex + 1) != '/'
+        || url.charAt(schemeEndIndex + 2) != '/') {
+      // has no authority component
+      return url;
+    }
+
+    // look for the end of the authority component:
+    //   '/', '?', '#' ==> start of path
+    int index;
+    for (index = schemeEndIndex + 3; index < len; index++) {
+      char c = url.charAt(index);
+      if (c == '/' || c == '?' || c == '#') {
+        break;
+      }
+    }
+
+    int atIndex = url.lastIndexOf("@", index - 1);
```

Comment:
Wouldn't 
```suggestion
    int atIndex = -1;
    for (int index = schemeEndIndex + 3; index < len; index++) {
      char c = url.charAt(index);
      if (c == '@') {
        atIndex = index;
      }
      if (c == '/' || c == '?' || c == '#') {
        break;
      }
    }

```
work as well? You're already iterating through the string, no need to look up `@` again with `lastIndexOf()`

Reply:
Like this we as well would need to drop the check for `atIndex==len` again if I'm not mistaken as the loop might break before right?

Reply:
No, I think it'd still be needed - `@` could be the last character in the string, and in that case you don't want to modify the url.